### DISCUSSION TITLE
Can we cross build `iree-run-module` for RISCV architecture?

### DISCUSSION THREAD

**Discussion by FlintWangacc** [iree-org/iree#17881] [General]
I am trying to run a model on RISCV devices. But the RISCV enviornment has limited resources so I can't build whole iree toolchain.
Is the following way possible?
1. build a model through iree toolchain to generate .vmfb file on server and copy to RISCV device.
2. cross build `iree-run-module` on server and copy to RISCV device.
3. Run iree-run-module

I understand how to build .vmfb file. Is there any way to cross build `iree-run-module`

**Comment by ScottTodd** [comment]
Yep, the runtime is designed to be cross compiled for all sorts of platforms. See https://iree.dev/building-from-source/riscv/#configure-and-build for RISC-V.

This could also be helpful to run or reference: https://github.com/iree-org/iree/blob/main/build_tools/cmake/build_riscv.sh. Here is a recent run of that script on our CI where you can see the variables fully expanded and other logs: https://github.com/iree-org/iree/actions/runs/9900554673/job/27351890948#step:7:129
